scripts/zkai_attention_kv_softmax_denominator_rounding_edge_corpus_gate.py

```
@@ -0,0 +1,488 @@
+    mutated_source = copy.deepcopy(source_payload)
+    mutated_fused = copy.deepcopy(fused_envelope)
+    mutate_source_remainder_equal_denominator(mutated_source)
+    mutated_fused["source_input"] = mutated_source
+    record(
+        "fused_matching_source_remainder_equal_denominator",
+        "fused",
+        lambda: fused_gate.validate_fused_envelope(mutated_fused, mutated_source, run_native=False),
+    )
+
+    if tuple(result["name"] for result in results) != ROUTE_MUTATION_NAMES:
+        raise SoftmaxEdgeCorpusGateError("route mutation order drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)